### PR TITLE
refactoring cbv tests

### PR DESCRIPTION
**Pull request by DawnOfHell**
taking tests that were written without using many of pytests abilities and refactoring it (run time was shorten from 0.15 seconds to ~0.12/0.11 seconds)

### CODE REVIEW

File: tests/test_cbv.py [hunk 1]
```diff
@@ -6,44 +6,67 @@
...
-        response_1 = client.get("/")
-        assert response_1.status_code == 200
-        assert response_1.json() == expected_response
+        return
 
-        response_2 = client.get("/sum")
-        assert response_2.status_code == 200
-        assert response_2.content == b"3"
+    def test_response_models(self, cbv_router: None,
+                             client: TestClient) -> None:
```
Comment (yuval9313) [inline tests/test_cbv.py:52]: Why has this become 2 lines all of a sudden?

File: tests/test_cbv.py [hunk 2]
```diff
@@ -6,44 +6,67 @@
...
+        response_home = client.get("/")
+        assert response_home.status_code == SUCCESS_CODE
+        assert response_home.json() == self.EXPECTED_RESPONSE
 
-    def test_dependencies(self, router: APIRouter) -> None:
+        sum_response = client.get("/sum")
+        assert sum_response.status_code == SUCCESS_CODE
+        assert sum_response.content == THREE_AS_BYTES
```
Comment (yuval9313) [inline tests/test_cbv.py:59]: Wouldn't it be better that test cases would differ?
e.g. 
```
def test_string_response():
    ...

def test_int_response():
    ....
```

Comment (yuval9313) [inline tests/test_cbv.py:59]: Would `assert sum_response.json() == 3` work?

File: tests/test_cbv.py [hunk 3]
```diff
@@ -56,12 +79,39 @@ def __init__(self, two: int = Depends(dependency_two)):
...
 
-        client = TestClient(router)
+        return
+
+    def test_dependencies(self, cbv_router: None, client: TestClient) -> None:
         response = client.get("/")
-        assert response.status_code == 200
-        assert response.content == b"3"
+        assert response.status_code == SUCCESS_CODE
+        assert response.content == THREE_AS_BYTES
```
Comment (yuval9313) [inline tests/test_cbv.py:87]: `assert response.json() == 3` maybe?

File: tests/test_cbv.py [hunk 4]
```diff
@@ -56,12 +79,39 @@ def __init__(self, two: int = Depends(dependency_two)):
             def int_dependencies(self) -> int:
                 return self.one + self.two
 
-        client = TestClient(router)
+        return
+
+    def test_dependencies(self, cbv_router: None, client: TestClient) -> None:
```
Comment (yuval9313) [inline tests/test_cbv.py:84]: Why would `cbv_router: None` how does it make sense?

File: tests/test_cbv.py [hunk 5]
```diff
@@ -70,27 +120,21 @@ class CBV:
...
                 return hasattr(self, "class_var")
 
-        client = TestClient(router)
+        return
+
+    def test_class_var(self, cbv_router: None, client: TestClient) -> None:
         response = client.get("/")
-        assert response.status_code == 200
+        assert response.status_code == SUCCESS_CODE
         assert response.content == b"false"
```
Comment (yuval9313) [inline tests/test_cbv.py:128]:
```suggestion
        assert response.json() == False

```

File: tests/test_cbv.py [hunk 6]
```diff
@@ -70,27 +120,21 @@ class CBV:
...
-            def get_any_path(self) -> int:  # Alphabetically before `get_test`
-                return 2
-
-        client = TestClient(router)
-        assert client.get("/test").json() == 1
-        assert client.get("/any_other_path").json() == 2
+class TestMultipleTests(AbstractTestRouter):
+    def setup_method(self):
+        self.item = "abc"
+        self.num_item = "1"
```
Comment (yuval9313) [inline tests/test_cbv.py:134]: `num_item` of type `str`?

**Review by yuval9313 — CHANGES_REQUESTED** [review]

File: tests/test_cbv.py [hunk 7]
```diff
@@ -6,44 +6,67 @@
 
 from fastapi_restful.cbv import cbv
 
+ONE = 1
+TWO = 2
```
[the remark this answers is not in the document]
Comment (DawnOfHell) [reply tests/test_cbv.py:10]: should they be named difrrent then? the qestion is. why do we keep using "1" and "2" as hard-coded variables? 
It's not a better solution- in my opinion the vriables we use as placeholders should be more indicative anyway (and maybe difrrent from test to test).

File: tests/test_cbv.py (hunk 1 above)
Comment (DawnOfHell) [reply tests/test_cbv.py:52]: line over 90 chars

File: tests/test_cbv.py (hunk 4 above)
Comment (DawnOfHell) [reply tests/test_cbv.py:84]: it returns None the fixture onlt initalizes the router and there is no need to return it as we don't use the object itself in the test

File: tests/test_cbv.py (hunk 5 above)
Comment (DawnOfHell) [reply tests/test_cbv.py:128]: does not pass b"false" object is the object returned while decoding it makes in a "false" string- there is no way to convert it to true boolean

File: tests/test_cbv.py (hunk 3 above)
Comment (DawnOfHell) [reply tests/test_cbv.py:87]: also does not work - it returns b'3' and decoding it retuns "3" as a string- it is possible but we will need to do some manipulation on the respose - and I  am not sure it is favorable as we want to check the response as is

File: tests/test_cbv.py (hunk 6 above)
Comment (DawnOfHell) [reply tests/test_cbv.py:134]: str is for the asser, when getting the custom path you compare 1 to "1" - we can convert it with int(), but do we want to manipulate the response from the router?

File: tests/test_cbv_base.py [hunk 8]
```diff
@@ -1,88 +1,137 @@
...
 
-    client = TestClient(app)
-    response_1 = client.post("/", params={"x": 1}, json={})
-    assert response_1.status_code == 200
-    assert response_1.content == b"4"
 
-    response_2 = client.get("/classvar")
-    assert response_2.status_code == 200
-    assert response_2.content == b"false"
+class TestCbv(AbstractTest):
```
Comment (yuval9313) [inline tests/test_cbv_base.py:29]: Very vague of you to call it `TestCbv`

File: tests/test_cbv_base.py [hunk 9]
```diff
@@ -1,88 +1,137 @@
...
+            def get(self, item_id: str) -> str:
+                return item_id
 
-    assert client.get("/items/1").json() == {"item_path": "1"}
-    assert client.get("/items").json() == []
+        return TestCBV
 
+    @staticmethod
+    @pytest.fixture(autouse=True)
+    def add_resource(cbv_resource: object, api: Api):
```
Comment (yuval9313) [inline tests/test_cbv_base.py:88]: I was sooooo confused from this
What had happened here? this is not commonly used and therefore not needed a separated fixture 
rest easy to merge this fixture with the one setting up application as they are the same

File: tests/test_cbv_base.py [hunk 10]
```diff
@@ -1,88 +1,137 @@
...
+    def cbv_resource(self):
+        class RootHandler(Resource):
+            @set_responses({}, response_class=PlainTextResponse)
+            def get(self) -> str:
+                return TestDifferentResponseModule.RESPONSE
+
+        return RootHandler
+
+    @pytest.fixture(autouse=True)
+    def add_resource(self, api: Api, cbv_resource: object):
```
Comment (yuval9313) [inline tests/test_cbv_base.py:133]: Fixture not needed

File: tests/test_cbv_base.py [hunk 11]
```diff
@@ -1,88 +1,137 @@
...
+                if item_path:
+                    return {"item_path": item_path}
+                return []
 
-    client = TestClient(app)
+        return RootHandler
 
-    assert client.get("/check").text == "Done!"
+    @pytest.fixture(autouse=True)
+    def add_resource(self, cbv_resource: object, api: Api):
```
Comment (yuval9313) [inline tests/test_cbv_base.py:109]: Fixture not needed

**Review by yuval9313 — CHANGES_REQUESTED** [review]

File: tests/test_cbv.py [hunk 12]
```diff
@@ -1,67 +1,118 @@
...
+            return TestDependencies.SECOND
 
         @cbv(router)
         class CBV:
-            one: int = Depends(dependency_one)
+            dataclass_defined_dependency: int = Depends(dependency_one)
 
             def __init__(self, two: int = Depends(dependency_two)):
-                self.two = two
+                self.init_defined_dependency = two
```
[the remark this answers is not in the document]
Comment (DawnOfHell) [reply tests/test_cbv.py:77]: so why is it in int deps...? that method name was not changed it this mr .....

**Pull request closed without merging**